pkg/hive/hive.go

```diff
@@ -134,16 +136,22 @@ type defaults struct {
 
 	Flags       *pflag.FlagSet
 	Lifecycle   Lifecycle
+	Health      cell.Health
 	Logger      logrus.FieldLogger
 	Shutdowner  Shutdowner
 	InvokerList cell.InvokerList
 }
 
 func (h *Hive) provideDefaults() error {
+	h.lifecycle.Append(Hook{OnStop: func(ctx HookContext) error {
```

Comment:
I'd instantiate `cell.Health` locally here since `h.healthStatus` doesn't need to be used by `Hive` directly. And perhaps for clarity we could have `NewHealthProvider` take `Lifecycle` as argument and thus we can just call `h.container.Provide(cell.NewHealthProvider)`.